stone/backends/python_rsrc/stone_base.py

```diff
@@ -34,8 +34,53 @@ def __deepcopy__(self, memo):
         # disable copying so we can do identity comparison even after copying stone objects
         return self
 
+
 NOT_SET = NotSet()  # dummy object to denote that a field has not been set
 
+NO_DEFAULT = object()
+
+
+class Attribute(object):
+    __slots__ = ("name", "default", "nullable", "user_defined")
+
+    def __init__(self, name, nullable=False, user_defined=False):
+        # type: (typing.Text, bool, bool) -> None
+        self.name = name
+        self.nullable = nullable
+        self.user_defined = user_defined
+        # These should be set later, because of possible cross-references.
+        self.default = NO_DEFAULT
+
+    def __get__(self, instance, owner):
+        # type: (typing.Any, typing.Any) -> typing.Any
+        if instance is None:
+            return self
+        value = getattr(instance, "_{}_value".format(self.name))
```

Comment:
I wonder if it's worth precalculating `_{}_value` and storing it in an attribute? This would increase memory use but speed things up.

Reply:
Yeah, I wasn't sure about this. I decided to start with more memory saving option, we can always go back to pre-computing if performance will ever be an issue.

Reply:
Actually I don't think we need the original name, so I will just replace it with the formatted one.